### ISSUE TITLE
URL scheme with #

### ISSUE THREAD

**Issue by utsavdevsharma** [astriskit/miniature-parakeet#1]
hi @astriskit , request you to explain your thought process for the url scheme using `#` / for using the `HashRouter`

`http://localhost:3000/#/`
`http://localhost:3000/#/fav-posts`

are there any issues / limitations in using direct urls `http://localhost:3000/fav-posts`

**Comment by astriskit** [comment]
Hello @utsavdevsharma - this is to put client-side routing in to perspective of user. Also without a hash, browser will reload the whole application - like it is calling some server asset - which is not the case with the project.

A nice read at - https://stackoverflow.com/questions/51974369/hashrouter-vs-browserrouter

**Comment by astriskit** [comment]
Feel free to reopen in case there is something else related to it.

**Comment by utsavdevsharma** [comment]
@astriskit let me rephrase the question

Hash routing have multiple limitations. Direct urls are more compatible and makes the react app future ready.

Now when we setup a new app using create-react-app, it does not contain a Router. The official documentation suggests `react-router-dom` but do not recommend `hash` . Official documentation do lists the limitations of HashRouter

Using html 5 history api via react router, direct urls work fine. Server assets are not downloaded even if you navigate to direct url

I wanted to understand your thought process to prefer hashes over direct urls

**Comment by utsavdevsharma** [comment]
https://reacttraining.com/react-router/web/api/HashRouter

**Comment by astriskit** [comment]
Though I've not faced any issues working with it yet, but thank you for bringing it to my notice, may be, I'd take a look into it.

Also I didn't find "limitations of HashRouter" in official documentation. Feel free to post one, in case you know the link.
